care/utils/models/validators.py

```diff
@@ -41,3 +42,56 @@ def _extract_errors(
 
             message = str(error).replace("\n\n", ": ").replace("\n", "")
             container.append(ValidationError(message))
+
+
+@deconstructible
+class PhoneNumberValidator(RegexValidator):
+    """
+    Any one of the specified types of phone numbers are considered valid.
+
+    Allowed types:
+    - `mobile` (Indian XOR International)
+    - `indian_mobile` (Indian only)
+    - `international_mobile` (International only)
+    - `landline` (Indian only)
+    - `support` (Indian only)
+
+    Example usage:
+
+    ```
+    field = models.CharField(
+        validators=[PhoneNumberValidator(types=("mobile", "landline", "support"))])
+    )
+    ```
+    """
+
+    indian_mobile_number_regex = r"^(?=^\+91)(^\+91\d{10}$)"
```

Comment:
Indian mobile numbers only start with 7, 8 or 9. Should we consider adding that to the validation?

Reply:
Done.

with ref. to https://en.wikipedia.org/wiki/Mobile_telephone_numbering_in_India
I'm checking if starts with **6, 7, 8 or 9** instead.

Updated tests for these too.